release/ray_release/scripts/ray_bisect.py

```diff
@@ -62,20 +82,25 @@ def _trigger_test_run(test: Test, commit: str) -> None:
     )
 
 
-def _obtain_test_result(buildkite_step_key: str) -> bool:
-    outcome = None
+def _obtain_test_result(buildkite_step_keys: List[str]) -> Dict[str, str]:
+    outcomes = {}
     wait = 30
     total_wait = 0
-    while outcome not in ["passed", "hard_failed", "soft_failed"]:
+    while len(outcomes) != len(buildkite_step_keys):
```

Comment:
use `while True:` here, and have a check after the inner for loop, and break if `len(outcomes) == len(buildkite_step_keys)`

Currently, it will always wait for 30 seconds no matter what, even when in our unit tests..

also, maybe change `wait` to something like 5 ?